Sources/AppCoinsSDK/UI/Purchase/ViewModels/BottomSheetViewModel.swift

```diff
@@ -149,6 +149,9 @@ internal class BottomSheetViewModel : ObservableObject {
     
     // User presses buy button
     internal func buy() {
+        
+        self.recordIntentClick(paymentMethod: TransactionViewModel.shared.paymentMethodSelected?.name ?? "")
```

Comment:
Just use the AnalyticsUseCases.shared.recordPurchaseIntent directly here. No need to create a new method on the ViewModel